### ISSUE TITLE
Update fails due to Nginx timeout

### ISSUE THREAD

**Issue by Altair31** [nextcloud/server#21375]
I've faced this problem for a long time now, but always find a more or less usable workaround.

I have a Nextcloud on a quite slow machine with a large storage.
My files take enough place so Nextcloud updater take a long time to backup.
I first drastically increase timeouts on Nginx settings but now it takes more than an hour just for the backup. I don't want to increase more my timeout settings.

Could your web-based updater periodically send data to keep the connection alive between PHP-FPM et Nginx ?

Now I'm forced to use the command-line updater just for this reason.

Thanks

**Comment by kesselb** [comment]
Thanks for reporting a issue :+1: 

For questions and discussions please visit https://help.nextcloud.com. If you want to report a bug or request a feature use the provided template. 

The updater repository / code is here: https://github.com/nextcloud/updater. I can move the issue over there if you want. 

> My files take enough place so Nextcloud updater take a long time to backup.

The backup step does not backup the user files.

> Could your web-based updater periodically send data to keep the connection alive between PHP-FPM et Nginx ?

Technically that is possible but requires to rewrite a lot of the updater code. 

> Now I'm forced to use the command-line updater just for this reason.

That's great! The cli updater is the fastest and most reliable (because no web server is involved) way to upgrade nextcloud. Run `php updater.phar` and restart php-fpm afterwards and you are done.

I can move the issue over to nextcloud/updater if you want but or otherwise close it.

**Comment by Altair31** [comment]
> For questions and discussions please visit https://help.nextcloud.com. If you want to report a bug or request a feature use the provided template.

Yes this was a feature request as there is no bug detected in the updater, just a wrong integration with Nginx (and may be other servers that use php-fpm).
If you want me to reformat my request message, please point me the template as I had not found it.

> The updater repository / code is here: https://github.com/nextcloud/updater. I can move the issue over there if you want.

Yes, please.

> > My files take enough place so Nextcloud updater take a long time to backup.
> 
> The backup step does not backup the user files.

Ok, then it should be addons that should make the backup step very long.

> > Could your web-based updater periodically send data to keep the connection alive between PHP-FPM et Nginx ?
> 
> Technically that is possible but requires to rewrite a lot of the updater code.

I'm completely aware and I'm not asking you to do it for the next major.
